### PR TITLE
 refactor(@angular-devkit/core): remove deprecated terminal utils

### PR DESCRIPTION
**Pull request by alan-agius4**
BREAKING CHANGE:

Deprecated `@angular-devkit/core` terminal API has been removed. Use 3rd party packages like chalk, colors or ansi-colors instead.

**Note:** this changes doesn't effect application developers

### CODE REVIEW

File: packages/angular_devkit/core/node/cli-logger.ts [hunk 1]
```diff
@@ -6,7 +6,8 @@
  * found in the LICENSE file at https://angular.io/license
  */
 import { filter } from 'rxjs/operators';
-import { logging, terminal } from '../src';
+import { logging } from '../src';
+import { colors as terminalColors } from './color';
```
Comment (alan-agius4) [inline packages/angular_devkit/core/node/cli-logger.ts:10]: I am actually inclined to remove colors from here and rely on consumers to pass the colors via the colors parameter when colors are desired.

Comment (clydin) [reply packages/angular_devkit/core/node/cli-logger.ts:10]: Sounds good.

File: packages/angular/cli/utilities/color.ts [hunk 2]
```diff
@@ -8,18 +8,16 @@
...
 
+type AnsiColors = typeof ansiColors;
+
 // Typings do not contain the function call (added in Node.js v9.9.0)
-export const supportsColor =
+const supportsColor =
   process.stdout instanceof WriteStream &&
   ((process.stdout as unknown) as { getColorDepth(): number }).getColorDepth() > 1;
 
-export function removeColor(text: string): string {
```
Comment (clydin) [inline packages/angular/cli/utilities/color.ts:16]: This was added due to the `unstyle` method from the package not working when colors were disabled.
https://github.com/doowb/ansi-colors/blob/a4794363369d7b4d1872d248fc43a12761640d8e/index.js#L38

**Comment by alan-agius4** [comment]
@clydin PTAL

**Review by clydin — APPROVED** [review]

**Pull request merged**